refactor/my_location_display

BarAgam opened the pull request:
Self and other users location displaying over the map.

gRaytan committed:
added: single timer for users location updates. (+4 squashed commit)

Squashed commit:

[4ca3177] add symbols and showing other users on map (in app module)

need to:
add timer for stale users.
show self user.

[0bf0b42] added: user entity to domain and data.

need to:
add symbolizer to user entity in app
add timer to make user enities stale (in domain)

[2762159] first commit

[01f6f58] Fix merge errors

gRaytan committed:
Fix merge errors

gRaytan committed:
first commit

teamagam1 committed:
added: user entity to domain and data.

need to:
add symbolizer to user entity in app
add timer to make user enities stale (in domain)

teamagam1 committed:
add symbols and showing other users on map (in app module)

need to:
add timer for stale users.
show self user.

barmako committed:
Implemented self location map display by interactor.
Interactor isn't executing atm.

barmako committed:
Fixed rebase issues.

barmako committed:
user location displaying wired.

teamagam1 committed:
rebase fixes

barmako committed:
Display user location finished.

review-ninja commented:
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/117" target="_blank"><img src="https://app.review.ninja/52010471/pull/117/badge" alt="ReviewNinja"/>

teamagam1 committed:
some CR fixes:

* naming and code conventions.
* LocationDataRepository is now singleton!
* some cleaning
* display my location interactor - uses our EntitiesRepository.

questions:
1. why do we need GGLocationService? and it's locationStreamInteractor.
2. why don't we use launcher activity for all the initiations of the repositories?
